edgecast/resources/cps/certificate.go

```diff
@@ -47,31 +47,61 @@ func ResourceCertificateCreate(
 		return diag.Errorf("failed to load configuration")
 	}
 
-	cpsService, err := buildCPSService(**config)
+	ns, errs := ExpandNotificationSettings(d.Get("notification_setting"))
+
+	svc, err := buildCPSService(**config)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	certificateObj, errs := ExpandCertificate(d)
+	cert, errs := ExpandCertificate(d)
 	if len(errs) > 0 {
 		return helper.DiagsFromErrors(errs)
 	}
 
 	// Call create certificate API
-	params := certificate.NewCertificatePostParams()
-	params.Certificate = certificateObj
+	cparams := certificate.NewCertificatePostParams()
+	cparams.Certificate = cert
+
+	cresp, err := svc.Certificate.CertificatePost(cparams)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	//ns, errs := ExpandNotificationSettings(d.Get("notification_settings"))
 
-	resp, err := cpsService.Certificate.CertificatePost(params)
+	if len(errs) > 0 {
+		return helper.DiagsFromErrors(errs)
```

Comment:
We should SetId("") here. There are also a few other places in this function we should be as well if you want to add them..